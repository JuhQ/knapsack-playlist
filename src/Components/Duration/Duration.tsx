import React from "react";

interface Props {
  seconds: number;
}

const Duration: React.FC<Props> = ({ seconds }: Props) => {
  const minutes = Math.floor(seconds / 60);
  const secondsCounted = seconds - minutes * 60;

  const minutesString =
    minutes > 0 ? `${minutes} minute${minutes !== 1 ? "s" : ""}` : "";

  const secondsString =
    secondsCounted > 0
      ? `${secondsCounted} second${secondsCounted !== 1 ? "s" : ""}`
      : "";

  return (
    <>
      {minutesString} {secondsString}
    </>
  );
};

export default Duration;
