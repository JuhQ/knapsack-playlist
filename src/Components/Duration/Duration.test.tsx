import React from "react";
import renderer from "react-test-renderer";

import Duration from "./Duration";

describe("Duration", () => {
  const values = [0, 10, 11, 30, 60, 61, 90, 100, 120, 121, 600, 3600];

  values.forEach((time) => {
    it(`should render ${time}`, () => {
      const component = renderer.create(<Duration seconds={time} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
