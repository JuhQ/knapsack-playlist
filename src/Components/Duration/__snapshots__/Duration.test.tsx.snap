// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duration should render 0 1`] = `
Array [
  "",
  " ",
  "",
]
`;

exports[`Duration should render 10 1`] = `
Array [
  "",
  " ",
  "10 seconds",
]
`;

exports[`Duration should render 11 1`] = `
Array [
  "",
  " ",
  "11 seconds",
]
`;

exports[`Duration should render 30 1`] = `
Array [
  "",
  " ",
  "30 seconds",
]
`;

exports[`Duration should render 60 1`] = `
Array [
  "1 minute",
  " ",
  "",
]
`;

exports[`Duration should render 61 1`] = `
Array [
  "1 minute",
  " ",
  "1 second",
]
`;

exports[`Duration should render 90 1`] = `
Array [
  "1 minute",
  " ",
  "30 seconds",
]
`;

exports[`Duration should render 100 1`] = `
Array [
  "1 minute",
  " ",
  "40 seconds",
]
`;

exports[`Duration should render 120 1`] = `
Array [
  "2 minutes",
  " ",
  "",
]
`;

exports[`Duration should render 121 1`] = `
Array [
  "2 minutes",
  " ",
  "1 second",
]
`;

exports[`Duration should render 600 1`] = `
Array [
  "10 minutes",
  " ",
  "",
]
`;

exports[`Duration should render 3600 1`] = `
Array [
  "60 minutes",
  " ",
  "",
]
`;
